### ISSUE TITLE
Chef broken on windows when creating directory

### ISSUE THREAD

**Issue by Jeppesen-io** [mitchellh/packer#1317]
the chef-solo provisioner fails with 'mkdir -p <dirname>', we'll the -p option does not exist in windows.

Log excerpt from below:
```javascript
2014/07/02 18:27:57 ui: ^[[0;32m    qemu: Creating directory: c:/windows/temp^[[0m
2014/07/02 18:27:57 /opt/packer/bin/packer-builder-qemu: 2014/07/02 18:27:57 opening new ssh session
2014/07/02 18:27:57 /opt/packer/bin/packer-builder-qemu: 2014/07/02 18:27:57 starting remote command: mkdir -p 'c:/windows/temp'
2014/07/02 18:27:57 /opt/packer/bin/packer-builder-qemu: 2014/07/02 18:27:57 remote command exited with '1': mkdir -p 'c:/windows/temp'
2014/07/02 18:27:57 /opt/packer/bin/packer-builder-qemu: 2014/07/02 18:27:57 [INFO] RPC endpoint: Communicator ended with: 1
2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 [INFO] 78 bytes written for 'stdout'
2014/07/02 18:27:57 [INFO] RPC client: Communicator ended with: 1
2014/07/02 18:27:57 [INFO] RPC endpoint: Communicator ended with: 1
2014/07/02 18:27:57 /opt/packer/bin/packer-command-build: 2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 [INFO] 78 bytes written for 'stdout'
2014/07/02 18:27:57 [INFO] RPC client: Communicator ended with: 1
2014/07/02 18:27:57 [INFO] RPC endpoint: Communicator ended with: 1
2014/07/02 18:27:57 /opt/packer/bin/packer-command-build: 2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 /opt/packer/bin/packer-command-build: 2014/07/02 18:27:57 [INFO] 78 bytes written for 'stdout'
2014/07/02 18:27:57 /opt/packer/bin/packer-command-build: 2014/07/02 18:27:57 [INFO] RPC client: Communicator ended with: 1
2014/07/02 18:27:57 /opt/packer/bin/packer-command-build: 2014/07/02 18:27:57 [INFO] RPC endpoint: Communicator ended with: 1
2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 [INFO] 78 bytes written for 'stdout'
2014/07/02 18:27:57 [INFO] RPC client: Communicator ended with: 1
2014/07/02 18:27:57 [INFO] RPC endpoint: Communicator ended with: 1
2014/07/02 18:27:57 /opt/packer/bin/packer-provisioner-chef-solo: 2014/07/02 18:27:57 [INFO] 78 bytes written for 'stdout'
2014/07/02 18:27:57 /opt/packer/bin/packer-provisioner-chef-solo: 2014/07/02 18:27:57 [INFO] 0 bytes written for 'stderr'
2014/07/02 18:27:57 /opt/packer/bin/packer-provisioner-chef-solo: 2014/07/02 18:27:57 [INFO] RPC client: Communicator ended with: 1
^[[0;32m    qemu: ^[[?7l^[[H^[[2J^[[1;1H^[[37m^[[40mThe syntax of the command is incorrect.^[[2;1H^[[?7h^[[0m
2014/07/02 18:27:57 ui: ^[[0;32m    qemu: ^[[?7l^[[H^[[2J^[[1;1H^[[37m^[[40mThe syntax of the command is incorrect.^[[2;1H^[[?7h^[[0m
2014/07/02 18:27:57 /opt/packer/bin/packer-builder-qemu: 2014/07/02 18:27:57 Deleting floppy disk: /tmp/packer690948831
2014/07/02 18:27:57 ui: ^[[1;32m==> qemu: Deleting output directory...^[[0m
^[[1;32m==> qemu: Deleting output directory...^[[0m
2014/07/02 18:27:57 ui error: ^[[1;31mBuild 'qemu' errored: Error creating staging directory: Non-zero exit status.^[[0m

**Comment by kmeinster** [comment]
Same applies for the chef-client provisioner, maybe a switch/case for windows? (where no -p is done)

**Comment by kmeinster** [comment]
Create a dirty hack to actually fix this, I doubt a pull request will be merged (besides the fact, I havn't tested it against linux yet) but for a quick dirty fix the 2 commits below might help:

Adding some basic windows support:
 https://github.com/catwithboots/packer/commit/a5306963fae71ca8c058be1eeedbac8218800de1

Adding fileseperator (and little cleaning):
https://github.com/catwithboots/packer/commit/8b27e32238d73de8f575f724bf2c4fc2e7fddd55


After some testing and cleaning up (and asking someone with real GO language if this is actually OK'ish) I will PR it.

**Comment by AnilKumarPacker** [comment]
The Same Issue exists even for provisioners of type "puppet-masterless".

2015/05/26 16:26:06 ui: [0;32m    null: Creating Puppet staging directory...[0m
2015/05/26 16:26:06 packer-builder-null.exe: 2015/05/26 16:26:06 opening new ssh session
2015/05/26 16:26:06 packer-builder-null.exe: 2015/05/26 16:26:06 starting remote command: mkdir -p 'E:/PackerTemplates/stagingdir'
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 remote command exited with '0': mkdir -p 'E:/PackerTemplates/stagingdir'
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 [INFO] RPC endpoint: Communicator ended with: 0
2015/05/26 16:26:07 [INFO] 0 bytes written for 'stderr'
2015/05/26 16:26:07 [INFO] 0 bytes written for 'stdout'
2015/05/26 16:26:07 [INFO] RPC client: Communicator ended with: 0
2015/05/26 16:26:07 [INFO] RPC endpoint: Communicator ended with: 0
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] RPC client: Communicator ended with: 0
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] 0 bytes written for 'stderr'
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] 0 bytes written for 'stdout'
2015/05/26 16:26:07 ui: [0;32m    null: Uploading manifests...[0m
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 opening new ssh session
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 starting remote command: mkdir -p 'E:/PackerTemplates/stagingdir/manifests'
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 remote command exited with '255': mkdir -p 'E:/PackerTemplates/stagingdir/manifests'
2015/05/26 16:26:07 packer-builder-null.exe: 2015/05/26 16:26:07 [INFO] RPC endpoint: Communicator ended with: 255
2015/05/26 16:26:07 [INFO] RPC client: Communicator ended with: 255
2015/05/26 16:26:07 [INFO] RPC endpoint: Communicator ended with: 255
2015/05/26 16:26:07 [INFO] 0 bytes written for 'stderr'
2015/05/26 16:26:07 [INFO] 0 bytes written for 'stdout'
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] RPC client: Communicator ended with: 255
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] 0 bytes written for 'stderr'
2015/05/26 16:26:07 packer-provisioner-puppet-masterless.exe: 2015/05/26 16:26:07 [INFO] 0 bytes written for 'stdout'
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui: [0;32m    null:[0m
2015/05/26 16:26:07 ui error: [1;31mBuild 'null' errored: Error uploading manifests: Error creating manifests directory: Non-zero exit status.[0m

**Comment by blairham** [comment]
+1

**Comment by kbarry-x** [comment]
I'm facing the same problem. I am unsure how to implement this fix. All I have is the exes. Would be great for this to make it into packer so I could use chef-client with windows.

**Comment by brumschlag** [comment]
+1

**Comment by kbarry-x** [comment]
Anychance we can get The PR reviewed? This is a significant blocker for us using packer with windows (Which we would be using a lot), in conjunctions with chef.

**Comment by c22** [comment]
Same issue exists in puppet-server and likely others, see #3850

**Comment by mwhooker** [comment]
I believe this was fixed in #1215

please re-open if I'm mistaken
